### DISCUSSION TITLE
How to change the admin logo and icon?

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#8303] [Q&A]
@agriffard had made [some great progress ](
https://github.com/OrchardCMS/OrchardCore/pull/5920)but the feature doesn't exist any more.

![image](https://user-images.githubusercontent.com/16193516/104509293-5eb73680-55e1-11eb-938a-d17172350a7a.png)

Any idea how to replace the admin logo and icon?

Many thanks!

**Comment by hishamco** [comment]
You can do it from the Admin Settings

**Reply by orchardc** [reply]
Where? I can't see it. Also agriffard just said it hasn't been merged.

**Reply by hishamco** [reply]
@agriffard I think you did this, right?

**Reply by hishamco** [reply]
@orchardc I thought @agriffard PR #5920 has been merged ;)

**Comment by Skrypt** [comment]
Quick workaround ; use your own admin theme.

**Reply by orchardc** [reply]
Thanks @Skrypt but how? I can think of maybe add the TheAdmin theme source code as a dependence of my project?
Can I just apply a minimal theme with this change only? I don't want to break some look&feel already built (without using recipes...

**Reply by Skrypt** [reply]
You can create a Admin Theme that inherits from TheAdmin. Then you create your own Layout.cshtml file in a Views folder and change what you need in there. On your Manifest.cs file of your custom admin theme you use the property "BaseTheme" from the ThemeAttribute class.

```C#
using OrchardCore.DisplayManagement.Manifest;

[assembly: Theme(
    Name = "My Admin Theme",
    BaseTheme = "TheAdmin",
    Author = "The Orchard Team",
    Website = "http://orchardproject.net",
    Version = "2.0.0",
    Description = "Your custom admin theme.",
    Tags = new[] { "admin" }
)]

```

And of course your custom admin theme project must have a different assembly name than "TheAdmin".

**Reply by orchardc** [reply]
I think I'm closed.

So I already have a custom theme, now I added    `BaseTheme = "TheAdmin",` in the Manifest.cs file

Then I updated the Layout.cshtml file and added 
` <style>    .ta-navbar-brand { background-image: url("/MyTheme/logo.png") !important; }</style>`

Then I run the website, go to the Admin panel, reset the Site Theme, then apply MyTheme (which inherits from TheAdmin now). 

But nothing has changed...The green logo is still there. 
![image](https://user-images.githubusercontent.com/16193516/104628021-a4323d00-568f-11eb-88a3-f4ad2e6ea547.png)

**Reply by Skrypt** [reply]
Your style won't work because it's an actual `<img>` that is set inside this `<div class="ta-navbar-brand">`.
What you need to do is change the src of the `<img src="/TheAdmin/logo.png" alt="@Site.SiteName" />`.

**Reply by jtkech** [reply]
Which now is `<img src=@Url.Content("~/TheAdmin/logo.png")` ;)

**Reply by orchardc** [reply]
This is solved now, thanks guys! :)

**Reply by orchardc** [reply]
Sorry still a problem. It looks like I have two themes now (1 for the front end UI and 1 for the Admin UI); but the CSS style impact each other... 

Maybe I took the wrong approach???

1. Just try to add the TheAdmin project (source code) into my solution and then set it as a project reference. 
2. In the Admin/Themes list, there are only 1 theme that can be set as current; but preivious the default TheAdmin theme is also applied (together with my theme).

So what is the right approach to add TheAdmin theme into my solution which already is a theme?

**Reply by jtkech** [reply]
You don't need to include TheAdmin in your solution

As @Skrypt said, just use `BaseTheme = "TheAdmin"` in your theme manifest

**Reply by orchardc** [reply]
But that will impact my own layout (front end UI) if I added <img src=@Url.Content("~/TheAdmin/logo.png")>  in my layout file isn't it? @jtkech

**Reply by Skrypt** [reply]
No, you add a Layout.cshtml in your custom Admin Theme /view folder. Not the Layout.cshtml from your frontend theme. So this will affect only your admin theme.

**Reply by orchardc** [reply]
Sorry I must misunderstood you. So currently I have two projects in the solution, the web project and the theme project (for the front end UI). Are you suggesting to have a third project or not?

**Reply by Skrypt** [reply]
Yes, the third project will be your custom admin theme that has only a Manifest.cs file and a /Views/Layout.cshtml file.
Then revert what your changed in your custom frontend theme. Because these are 2 separate themes.
Also, don't forget on that admin theme to reference
`<ProjectReference Include="..\..\OrchardCore\OrchardCore.Theme.Targets\OrchardCore.Theme.Targets.csproj" />`
And of course reference this new theme in your OrchardCore.Cms.Web.csproj so that it includes it.

**Reply by orchardc** [reply]
Thanks @Skrypt , where (which file) should we add this line ? Is it in the Custom Admin Theme's project file??

<ProjectReference Include="..\..\OrchardCore\OrchardCore.Theme.Targets\OrchardCore.Theme.Targets.csproj" />

**Reply by Skrypt** [reply]
In the CustomAdminTheme.csproj file yes.

Nuget Package solution : 
```
<PackageReference Include="OrchardCore.Theme.Targets" Version="$(OrchardCoreVersion)" />
```

Where $(OrchardCoreVersion) here is a var I'm using which can be replaced directly by a specific OC version like : 1.0.0-rc2-15765

Source Code solution : 
```
<ProjectReference Include="..\..\OrchardCore\OrchardCore.Theme.Targets\OrchardCore.Theme.Targets.csproj" />
```

**Reply by orchardc** [reply]
Got you! Thanks a lot !👍
